src/main/kotlin/com/terraformation/backend/tracking/db/PlantingSiteStore.kt

```diff
@@ -1718,9 +1718,114 @@ class PlantingSiteStore(
+                                .from(PLANTING_ZONE_POPULATIONS)
+                                .join(PLANTING_ZONES)
+                                .on(PLANTING_ZONE_ID.eq(PLANTING_ZONES.ID))
+                                .where(
+                                    PLANTING_ZONES.PLANTING_SITE_ID.eq(
+                                        PLANTING_SITE_POPULATIONS.PLANTING_SITE_ID))))
+                .execute()
+
+        log.info("Inserted $rowsInserted planting zone populations")
+      }
+
+      with(PLANTING_SUBZONE_POPULATIONS) {
+        val rowsInserted =
+            dslContext
+                .insertInto(
+                    PLANTING_SUBZONE_POPULATIONS,
+                    PLANTING_SUBZONE_ID,
+                    SPECIES_ID,
+                    TOTAL_PLANTS,
+                    PLANTS_SINCE_LAST_OBSERVATION)
+                .select(
+                    DSL.select(
+                            PLANTING_SUBZONES.ID,
+                            PLANTING_SITE_POPULATIONS.SPECIES_ID,
+                            PLANTING_SITE_POPULATIONS.TOTAL_PLANTS,
+                            PLANTING_SITE_POPULATIONS.PLANTS_SINCE_LAST_OBSERVATION)
+                        .from(PLANTING_SITE_POPULATIONS)
+                        .join(PLANTING_SUBZONES)
+                        .on(
+                            PLANTING_SITE_POPULATIONS.PLANTING_SITE_ID.eq(
+                                PLANTING_SUBZONES.PLANTING_SITE_ID))
+                        .whereNotExists(
+                            DSL.selectOne()
+                                .from(PLANTING_SUBZONE_POPULATIONS)
+                                .join(PLANTING_SUBZONES)
+                                .on(PLANTING_SUBZONE_ID.eq(PLANTING_SUBZONES.ID))
+                                .where(
+                                    PLANTING_SUBZONES.PLANTING_SITE_ID.eq(
+                                        PLANTING_SITE_POPULATIONS.PLANTING_SITE_ID))))
+                .execute()
+
+        log.info("Inserted $rowsInserted planting subzone populations")
+      }
     }
   }
 
+  private fun migrateSimplePlantingSitePlantings() {
+    // Outplanting withdrawals to simple planting sites result in rows in the plantings table
+    // with the planting site ID populated but the planting subzone ID set to null. Withdrawals to
+    // detailed planting sites are required to have a subzone ID. So we want to fill in the subzone
+    // ID for any plantings that have null subzone IDs but where the planting site has a subzone,
+    // since those will be the sites whose zones and subzones we've just created.
+    val rowsUpdated =
+        dslContext
+            .update(PLANTINGS)
+            .set(
+                PLANTINGS.PLANTING_SUBZONE_ID,
+                DSL.select(PLANTING_SUBZONES.ID)
+                    .from(PLANTING_SUBZONES)
+                    .where(PLANTING_SUBZONES.PLANTING_SITE_ID.eq(PLANTINGS.PLANTING_SITE_ID)))
```

Comment:
Is it possible that this could have multiple subzones, thus breaking this query? Or is that not possible because it was a simple planting site before? (I don't know much about simple planting sites)

Reply:
That _shouldn't_ be able to happen, but I'll add a guard against it anyway; good call.